lib/llm/provider.rb

```diff
@@ -31,6 +31,10 @@ def complete(prompt)
       raise NotImplementedError
     end
 
+    def chat(messages)
```

Comment:
It is a good practice for the method signature on the superclass to match the method signature on the subclass as well. So I think this should be:
```ruby
def chat(prompt, **params)
  # ...
end
```

Reply:
@0x1eef noted!